Tasks/GoToolV0/Tests/L0CachedMicrosoft.ts

```diff
@@ -0,0 +1,65 @@
+import tmrm = require('azure-pipelines-task-lib/mock-run');
+import path = require('path');
+
+let taskPath = path.join(__dirname, '..', 'gotool.js');
+let tmr: tmrm.TaskMockRunner = new tmrm.TaskMockRunner(taskPath);
+
+// Set inputs for cached Microsoft Go version
+tmr.setInput('version', '1.25.0');
+tmr.setInput('goDownloadBaseUrl', 'https://aka.ms/golang/release/latest');
+
+// Mock tool lib functions
+tmr.registerMock('azure-pipelines-tool-lib/tool', {
+    findLocalTool: function(toolName: string, version: string) {
+        console.log(`Found cached tool: ${toolName} version ${version}`);
+        // Return cached path to simulate found Microsoft build
+        return '/mock/cache/go-aka/1.25.0';
```

Comment:
Doesn't this need to be a fully qualified version in order to be cached? It seems to me that the partial version 1.25.0 should request the manifest so that it can discover that the latest version is 1.25.0-1 (in this mock), and find 1.25.0-1 in the cache.